src/main/java/core/basesyntax/Lottery.java

```diff
@@ -0,0 +1,14 @@
+package core.basesyntax;
+
+public class Lottery {
+    public Ball getRandomBall() {
+        ColorSupplier colorSupplier = new ColorSupplier();
+        String randomColor = colorSupplier.getRandomColor();
+
+        Ball ball = new Ball();
+        ball.setColor(randomColor);

```

Comment:
INFO: You can replace setter calls with just a single line of two-argument constructor with which you can provide all necessary dependencies.